src/app.ts

```diff
@@ -22,28 +23,53 @@ log.info('rocketchat2matrix starts.')
  * @param entity The Entity with it's file name and type definitions
  */
 async function loadRcExport(entity: Entity) {
+  const concurrency = parseInt(process.env.CONCURRENCY_LIMIT || '50')
+  const user_queue: RcUser[] = []
+  const room_queue: RcRoom[] = []
+  const messages_per_room: Map<string, RcMessage[]> = new Map()
+
   const rl = new lineByLine(`./inputs/${entities[entity].filename}`)
 
   let line: false | Buffer
   while ((line = rl.next())) {
     const item = JSON.parse(line.toString())
     switch (entity) {
```

Comment:
Yes, threadmessages are handled just like any message, so the sequence is the same as from the RC export.
I converted the variables to camelCase. Thanks!

Reply:
As the function is called 3 times, each with one entity type, it would check 3 times and loop within it. We also don't need to check non-affected queues, then.

Reply:
In https://github.com/verdigado/rocketchat2matrix/pull/31/commits/9fa8e033ed0686507e4c901caa0aaadc18fd345c,
https://github.com/verdigado/rocketchat2matrix/pull/31/commits/64935fd815b736a1b42f6dc1cb1f5d73c301722f
 and https://github.com/verdigado/rocketchat2matrix/pull/31/commits/7a7fdecb73c49f830c68785933f0962007162730 :

I moved the switch outside of the loop. For users and rooms, the loop is handled by the PromisePool, so no need to use queues anymore. For messages, to maintain the sequence, I kept the two-step populating of a Map and loading.
Because `PromisePool` loops over an iterator, I added a wrapper to turn `lineByLine` into an iterator (its `.next()` method made me think it was implemented as an iterator, but it isn't).

I feel the message handling can still be improved, maybe avoiding the first step of loading. To take care of https://github.com/verdigado/rocketchat2matrix/issues/3#issuecomment-2236672501, cut the messages in batches or something.

Reply:
> I feel the message handling can still be improved, maybe avoiding the first step of loading. To take care of [#3 (comment)](https://github.com/verdigado/rocketchat2matrix/issues/3#issuecomment-2236672501), cut the messages in batches or something.

I added batches for the message handling in [bee1c2f](https://github.com/verdigado/rocketchat2matrix/pull/31/commits/bee1c2fc192d313cb5cefc6ff65f3d1e8246cfdb).
Tested with:
```sh
./reset.sh && npm run compile && systemd-run --scope -p MemoryMax=75M -p MemorySwapMax=1K --user node dist/app.js 
```